### DISCUSSION TITLE
how to solve akvorado traffick is not the same as in the router

### DISCUSSION THREAD

**Discussion by Ndaknah** [akvorado/akvorado#1552] [Q&A]
Hello Everyone,

i have question 
I have real traffic on the router 400gbps but in akvorado it only shows 80gbps.
between the server and the router sampling rate is the same
does anyone have the same problem

error in server:
 ` curl -s http://127.0.0.1:8081/api/v0/inlet/metrics | grep '^akvorado_inlet' | grep _errors
akvorado_inlet_core_flows_errors_total{error="SNMP cache miss",exporter="10.10.10.1"} 140
akvorado_inlet_core_flows_errors_total{error="sampling rate missing",exporter="10.10.10.1"} 1896
akvorado_inlet_flow_decoder_errors_total{name="netflow"} 8763`

`curl -s http://127.0.0.1:8081/api/v0/inlet/flows\?limit=1 | grep SamplingRate
    "SamplingRate": 1024,`

set in ruter juniper
`set forwarding-options sampling input rate 1024
set forwarding-options sampling instance NFSEN1 input rate 1024`

![Screenshot 2024-12-11 161752](https://github.com/user-attachments/assets/d620c8d1-d59d-4c9d-8fe3-948dd7bb55a0)

**Comment by fatherfbi** [comment]
Show
`> show chassis fpc`
and
`> show configuration chassis | display set`
Do you use logical systems in your configuration?

**Reply by Ndaknah** [reply]
`show chassis fpc 
                     Temp  CPU Utilization (%)   CPU Utilization (%)  Memory    Utilization (%)
Slot State            (C)  Total  Interrupt      1min   5min   15min  DRAM (MB) Heap     Buffer
  0  Online            30      1          0        1      1      1    32768      55          1`

`show configuration chassis | display set 
set chassis redundancy routing-engine 0 master
set chassis redundancy routing-engine 1 backup
set chassis aggregated-devices ethernet device-count 100
set chassis fpc 0 pic 0 pic-mode 100G
set chassis fpc 0 pic 0 tunnel-services bandwidth 100g
set chassis fpc 0 pic 0 inline-services bandwidth 1g
set chassis fpc 0 sampling-instance NFSEN1
set chassis fpc 0 inline-services flex-flow-sizing
set chassis alarm management-ethernet link-down ignore
set chassis network-services enhanced-ip`

I don't use a logical system

**Reply by fatherfbi** [reply]
And
show services flow-monitoring | display set
show forwarding-options sampling | display set

**Reply by Ndaknah** [reply]
`set forwarding-options sampling input rate 1024
set forwarding-options sampling instance NFSEN1 input rate 1024
set forwarding-options sampling instance NFSEN1 family inet output flow-server 10.10.10.2 port 2055
set forwarding-options sampling instance NFSEN1 family inet output flow-server 10.10.10.2 version9 template NFSEN1
set forwarding-options sampling instance NFSEN1 family inet output inline-jflow source-address 10.10.10.1`

**Reply by fatherfbi** [reply]
and **show services flow-monitoring | display set**

**Reply by Ndaknah** [reply]
`display set 
set services flow-monitoring version9 template NFSEN1 ipv4-template`

**Reply by fatherfbi** [reply]
Delete your flow configuration and apply this one

set services flow-monitoring version9 template ipv4 flow-active-timeout 10
set services flow-monitoring version9 template ipv4 flow-inactive-timeout 10
set services flow-monitoring version9 template ipv4 template-refresh-rate packets 30
set services flow-monitoring version9 template ipv4 template-refresh-rate seconds 30
set services flow-monitoring version9 template ipv4 option-refresh-rate packets 30
set services flow-monitoring version9 template ipv4 option-refresh-rate seconds 30
set services flow-monitoring version9 template ipv4 ipv4-template
set services flow-monitoring version9 template ipv6 flow-active-timeout 10
set services flow-monitoring version9 template ipv6 flow-inactive-timeout 10
set services flow-monitoring version9 template ipv6 template-refresh-rate packets 30
set services flow-monitoring version9 template ipv6 template-refresh-rate seconds 30
set services flow-monitoring version9 template ipv6 option-refresh-rate packets 30
set services flow-monitoring version9 template ipv6 option-refresh-rate seconds 30
set services flow-monitoring version9 template ipv6 ipv6-template
set forwarding-options sampling instance AKVORADO input rate 1024
set forwarding-options sampling instance AKVORADO input max-packets-per-second 65535
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 port 2055
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 autonomous-system-type origin
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 source-address 10.10.10.1
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 version9 template ipv4
set forwarding-options sampling instance AKVORADO family inet output inline-jflow source-address 10.10.10.1
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 port 2055
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 autonomous-system-type origin
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 source-address 10.10.10.1
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 version9 template ipv6
set forwarding-options sampling instance AKVORADO family inet6 output inline-jflow source-address 10.10.10.1

set chassis fpc 0 sampling-instance AKVORADO

**Reply by Ndaknah** [reply]
whether ip 10.255.32.11 should be 10.10.10.1

**Reply by fatherfbi** [reply]
Update

**Reply by fatherfbi** [reply]
Has the situation changed?

**Reply by Ndaknah** [reply]
`show chassis fpc `
                     Temp  CPU Utilization (%)   CPU Utilization (%)  Memory    Utilization (%)
Slot State            (C)  Total  Interrupt      1min   5min   15min  DRAM (MB) Heap     Buffer
  0  Online            36      1          0        1      1      1    32768      52          1
  
`show configuration chassis | display set `
set chassis redundancy routing-engine 0 master
set chassis redundancy routing-engine 1 backup
set chassis aggregated-devices ethernet device-count 100
set chassis fpc 0 pic 0 pic-mode 100G
set chassis fpc 0 pic 0 tunnel-services bandwidth 100g
set chassis fpc 0 pic 0 inline-services bandwidth 1g
set chassis fpc 0 sampling-instance AKVORADO
set chassis fpc 0 inline-services flex-flow-sizing
set chassis alarm management-ethernet link-down ignore
set chassis network-services enhanced-ip

`show services flow-monitoring | display set `
set services flow-monitoring version9 template ipv4 flow-active-timeout 10
set services flow-monitoring version9 template ipv4 flow-inactive-timeout 10
set services flow-monitoring version9 template ipv4 template-refresh-rate packets 30
set services flow-monitoring version9 template ipv4 template-refresh-rate seconds 30
set services flow-monitoring version9 template ipv4 option-refresh-rate packets 30
set services flow-monitoring version9 template ipv4 option-refresh-rate seconds 30
set services flow-monitoring version9 template ipv4 ipv4-template
set services flow-monitoring version9 template ipv6 flow-active-timeout 10
set services flow-monitoring version9 template ipv6 flow-inactive-timeout 10
set services flow-monitoring version9 template ipv6 template-refresh-rate packets 30
set services flow-monitoring version9 template ipv6 template-refresh-rate seconds 30
set services flow-monitoring version9 template ipv6 option-refresh-rate packets 30
set services flow-monitoring version9 template ipv6 option-refresh-rate seconds 30
set services flow-monitoring version9 template ipv6 ipv6-template

`show forwarding-options sampling | display set` 
set forwarding-options sampling instance AKVORADO input rate 1024
set forwarding-options sampling instance AKVORADO input max-packets-per-second 65535
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 port 2055
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 autonomous-system-type origin
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 source-address 10.10.10.1
set forwarding-options sampling instance AKVORADO family inet output flow-server 10.10.10.2 version9 template ipv4
set forwarding-options sampling instance AKVORADO family inet output inline-jflow source-address 10.10.10.2
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 port 2055
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 autonomous-system-type origin
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 source-address 10.10.10.1
set forwarding-options sampling instance AKVORADO family inet6 output flow-server 10.10.10.2 version9 template ipv6
set forwarding-options sampling instance AKVORADO family inet6 output inline-jflow source-address 10.10.10.1

`show services flow-monitoring | display set` 
set services flow-monitoring version9 template ipv4 flow-active-timeout 10
set services flow-monitoring version9 template ipv4 flow-inactive-timeout 10
set services flow-monitoring version9 template ipv4 template-refresh-rate packets 30
set services flow-monitoring version9 template ipv4 template-refresh-rate seconds 30
set services flow-monitoring version9 template ipv4 option-refresh-rate packets 30
set services flow-monitoring version9 template ipv4 option-refresh-rate seconds 30
set services flow-monitoring version9 template ipv4 ipv4-template
set services flow-monitoring version9 template ipv6 flow-active-timeout 10
set services flow-monitoring version9 template ipv6 flow-inactive-timeout 10
set services flow-monitoring version9 template ipv6 template-refresh-rate packets 30
set services flow-monitoring version9 template ipv6 template-refresh-rate seconds 30
set services flow-monitoring version9 template ipv6 option-refresh-rate packets 30
set services flow-monitoring version9 template ipv6 option-refresh-rate seconds 30
set services flow-monitoring version9 template ipv6 ipv6-template

![WhatsApp Image 2024-12-13 at 10 26 36_ef9426d9](https://github.com/user-attachments/assets/164dab74-8e71-4cec-a4cc-008b6fd375f0)



even reduced traffic

**Reply by fatherfbi** [reply]
Have you enabled sampling on all the necessary ports?

**Reply by Ndaknah** [reply]
`show interfaces descriptions |count `
Count: 100 lines

`show configuration |display set|match samp |count `
Count: 131 lines

everything has been entered

**Reply by fatherfbi** [reply]
Check that you have this
```
set groups sampling interfaces <*> unit <*> family inet sampling input
set groups sampling interfaces <*> unit <*> family inet6 sampling input
```

And show me pls
`show configuration interfaces | display set | match sampling | count`

**Reply by Ndaknah** [reply]
```
set groups interfaces <> unit <> family inet sampling input      
error: syntax error: <*>
```

I can't use the command

```
show configuration interfaces | display set | match sampling | count 
Count: 118 lines
```

**Comment by fatherfbi** [comment]
Sorry, I'm out of ideas, in order to understand what's wrong, I need a full config of akvorado and your router (only fpc and interface)

**Reply by fatherfbi** [reply]
Let me know if the problem still occurs to you, I think I know what the cause might be.

**Reply by Ndaknah** [reply]
I'm still having the same problem, maybe you can help me again.

**Comment by zxyz** [comment]
Hello @Ndaknah,
are the counters you provided above at "error in server" rising or more or less stable?
The cache miss etc are I guess normal at startup but should not increase later on.

In the visualize screenshot you shared above, did you limit the displayed traffic to "internal" or did you remove the default `InIfBoundary = external` filter?
